### PR TITLE
[BugFix] Need local shuffle for the merged local agg

### LINKED ISSUE #14601: Results of 3/4 phase agg is wrong, when the second phase is pruned
### Steps to reproduce the behavior (Required)
```sql
CREATE TABLE `t1` (
  `x` int(11) NOT NULL COMMENT "",
  `y` int(11) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`x`)
COMMENT "OLAP"
PARTITION BY RANGE(`y`)
(PARTITION p1 VALUES [("1"), ("10")),
PARTITION p2 VALUES [("10"), ("20")),
PARTITION p3 VALUES [("20"), ("30")),
PARTITION p4 VALUES [("30"), ("40")),
PARTITION p5 VALUES [("40"), ("50")),
PARTITION p6 VALUES [("50"), ("60")),
PARTITION p7 VALUES [("60"), ("70")))
DISTRIBUTED BY HASH(`x`) BUCKETS 3
PROPERTIES (
"replication_num" = "1",
"colocate_with" = "group_test1"
);

insert into t1 values (1, 1), (2, 11), (3, 21), (4, 31), (5, 41), (6, 51);
insert into t1 values (1, 51), (2, 41), (3, 31), (4, 21), (5, 11), (6, 1);

select count(distinct x), sum(y) from t2;

-- 
```

### Expected behavior (Required)
```
+-------------------+--------+
| count(DISTINCT x) | sum(y) |
+-------------------+--------+
| 6                 | 312    |
+-------------------+--------+
1 row in set
```

### Real behavior (Required)
```
+-------------------+--------+
| count(DISTINCT x) | sum(y) |
+-------------------+--------+
| 12                | 312    |
+-------------------+--------+
```

### PR DESCRIPTION
**Pull request by ZiheLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14601.

## Problem Summary(Required) ：

As for decomposing `AggregateBlockingNode` to pipeline, a local shuffle will be interpolated, if the aggregation need finalize and has group by clauses.

However, as for the three/four-phase agg whose second phases is pruned, there is also `ScanNode -> (serialize) BlockingAgg`. Therefore, serialize BlockingAgg also need local shuffle.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [x] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|1|1|100.00%|[]|

**Review by satanson — DISMISSED** [review]

File: be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp [hunk 1]
```diff
@@ -228,14 +228,25 @@ std::vector<std::shared_ptr<pipeline::OperatorFactory>> AggregateBlockingNode::d
         return context->maybe_interpolate_local_shuffle_exchange(runtime_state(), ops, group_by_expr_ctxs);
     };
 
-    if (agg_node.need_finalize && !sorted_streaming_aggregate) {
-        // If finalize aggregate with group by clause, then it can be parallelized
-        if (has_group_by_keys) {
-            if (could_local_shuffle) {
+    if (!sorted_streaming_aggregate) {
+        // 1. Finalize aggregation:
+        //   - Without group by clause, it cannot be paralized and need local passthough.
```
Comment (LiShuMing) [inline be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp:233]: type `paralized ` -> `parallelized `

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalHashAggregateOperator.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**